libs/types-common/src/Data/Qualified.hs

```diff
@@ -127,6 +129,16 @@ foldQualified loc f g q
   | otherwise =
       g (qTagUnsafe q)
 
+pattern Local :: forall a x. forall. Local a -> (Local x, Qualified a)
```

Comment:
Minor point, but would it make sense to use an ad-hoc data type here instead of a pair? Pattern matching on a pair and getting `Local` or `Remote` values is a bit strange. Maybe something like:
```haskell
case q `RelativeTo` loc of
  Local x -> ...
  Remote x -> ...
```
might be a bit more readable than the equivalent with a pair.

Reply:
should it be `type RelativeTo = (,)`?

Reply:
I was thinking more `data RelativeTo a b = RelativeTo (Qualified a) (Local b)`.

Reply:
ahhh maybe it's fine actually, yeah, let's go with that :)